src/components/AppModal.vue

```diff
@@ -1,25 +1,47 @@
 <template>
   <transition name="modal">
     <div class="modal-mask">
-      <div class="modal-wrapper">
-        <section class="modal-container">
-          <h2 class="modal-header">
+      <div class="modal-wrapper" @click.self="close">
+        <div
+          class="modal-container"
+          role="dialog"
+          aria-labelledby="modalTitle"
+          aria-describedby="modalDescription"
+        >
+          <header id="modalTitle" class="modal-header">
             <slot name="header">default header</slot>
-          </h2>
+          </header>
 
-          <div class="modal-body">
+          <section class="modal-body" id="modalDescription">
             <slot name="body">default body</slot>
-          </div>
+          </section>
 
-          <button class="modal-button" @click="$emit('close')">Done</button>
-        </section>
+          <button type="button" aria-label="Close modal" class="modal-button" @click="close">Done</button>
+        </div>
       </div>
     </div>
   </transition>
 </template>
 
 <script>
-export default {};
+export default {
+  mounted() {
+    document.addEventListener("keydown", this.closeOnEsc);
```

Comment:
I'm excited to see stuff like this implemented in the new hooks / composable components API!